index.ts

```diff
@@ -53,12 +59,30 @@ export class HawkTracerClient {
             this._client = new NativeClient(option);
         }
         else {
-            this._client = new NativeClient(option.source, option.map_files);
+            this._client = new NativeClient(option.source, option.map_files, option.wait_for_server);
         }
     }
 
-    public start(): boolean {
-        return this._client.start();
+    public start(): Promise<boolean> {
+        const tryConnect = (resolve: any, reject: any) => {
+            try {
+                if (this._client.start()) {
+                    resolve(true);
```

Comment:
How about resolving to `this`? It'd be useful for the client to make the code a bit nicer.
`true` doesn't do anything anyway, do it?

Reply:
Done